### PR TITLE
Restructure README into docs

### PR DESCRIPTION
**Pull request by brendan-ward**
This moves most of the content out of the README file into a series of documentation files, and clarifies some of the content.

Built docs [here](https://pyogrio--28.org.readthedocs.build/en/28/)

(ignore CI build failures, those are unrelated)

### CODE REVIEW

File: docs/source/supported_formats.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Supported vector formats
+
+Support for reading and writing spatial data ultimately depends on what is
+available in your particular distribution of GDAL. GDAL supports reading from
+a wide number of vector file formats, and writing for a much smaller number.
+
+Please see the [list of drivers](https://gdal.org/drivers/vector/index.html) for
+more information.
+
+## Full and write support
```
Comment (martinfleis) [inline docs/source/supported_formats.md:10]:
```suggestion
## Full read and write support
```
```suggestion
## Full and write support
```

File: docs/source/install.md [hunk 2]
```diff
@@ -24,14 +24,22 @@ I/O support.
 
 ### PyPi
```
Comment (martinfleis) [inline docs/source/install.md:25]:
```suggestion
### PyPI
```
```suggestion
### PyPi
```

**Review by martinfleis — COMMENTED** [review]
Thanks! This looks great! Just two super minor suggestions.

**Comment by brendan-ward** [comment]
Thanks for the review @martinfleis - apologies for not getting to it sooner!

**Pull request merged**